File: example-android/src/main/java/com/komoju/android/ui/screens/store/FakeStoreUiState.kt
package com.komoju.android.ui.screens.store

import androidx.annotation.DrawableRes
import androidx.annotation.StringRes

data class FakeStoreUiState(val items: List<Item> = emptyList(), val isCreatingSession: Boolean = false, val error: String? = null)

data class Item(
    @DrawableRes val imageID: Int,
    @StringRes val name: Int,
    @StringRes val description: Int,
    val price: String,
    val model: String,
    @StringRes val color: Int,
    val isFavorite: Boolean = false,
)

File: example-ios/example-ios/screens/store/FakeStoreUiState.swift
import Foundation
import SwiftUI

struct FakeStoreUiState {
    var items: [Item] = []
    var isCreatingSession: Bool = false
    var error: String?
}

struct Item: Identifiable, Equatable {
    var id: UUID = .init()
    var imageResource: ImageResource
    var name: LocalizedStringKey
    var description: LocalizedStringKey
    var price: String
    var model: String
    var color: LocalizedStringKey
    var isFavorite: Bool = false
}
